configuration/src/main/java/io/airlift/configuration/ConfigurationFactory.java

```diff
@@ -472,7 +472,7 @@ private static <T> T newInstance(ConfigurationMetadata<T> configurationMetadata)
             if (e instanceof InvocationTargetException && e.getCause() != null) {
                 e = e.getCause();
             }
-            throw exceptionFor(e, "Error creating instance of configuration class [%s]", configurationMetadata.getConfigClass().getName());
+            throw exceptionFor(e, "Error creating instance of configuration class [%s], caused by %s: %s", configurationMetadata.getConfigClass().getName(), e.getClass().getSimpleName(), e.getMessage());
```

Comment:
The cause should already be recorded in the underlying cause of the exception, so this should be unnecessary. Is that not working as expected?

Reply:
@martint it's not. Guice will not display this cause when formatting errors

Reply:
This is due to a strange condition that if the underlying root cause is a single error, cause won't be rendered. I can point to the relevant code. This makes it hard to understand why configuration has failed to build due to a generic, non actionable error message

Reply:
> that if the underlying root cause is a single error, cause won't be rendered

Sounds like we should fix that.

Reply:
@martint within guice. We can copy whole formatting logic